airflow/secrets/cache.py

```diff
@@ -82,8 +82,7 @@ def get_variable(cls, key: str) -> str | None:
 
     @classmethod
     def save_variable(cls, key: str, value: str | None):
-        """saves the value for that key in the cache, if enabled"""
-        cls.init()  # ensure initialization has been done
```

Comment:
not doing the init automatically when a value is saved means we control when the cache is enabled.
In this case, it's only called in https://github.com/apache/airflow/pull/30259/files#diff-827b7a469438ffd1b172ae295134a84c8a914c9ea4c9ea2a7d3de1f1d5aa6bb6R1071
so it's only active when parsing dags.